Remove @borsboom maintainership

borsboom opened the pull request:
I haven't actively used Haskell in some time and am not really able to usefully respond to maintenance requirements for these packages.  Is there a way to find someone to take them over?

---

Checklist:
- [x] Meaningful commit message, eg `add my-cool-package` (please don't mention `build-constraints.yml`)
- [x] At least 30 minutes have passed since uploading to Hackage
- [x] If applicable, required system libraries are added to [02-apt-get-install.sh](https://github.com/commercialhaskell/stackage/blob/master/docker/02-apt-get-install.sh) or [03-custom-install.sh](https://github.com/commercialhaskell/stackage/blob/master/docker/03-custom-install.sh)
- [ ] (optional) Package is compatible with the latest version of all dependencies (Run `cabal update && cabal outdated`)
- [ ] (optional) Package have been verified to work with the latest nightly snapshot, e.g by running the [verify-package script](https://github.com/commercialhaskell/stackage/blob/master/verify-package)

The script runs virtually the following commands in a clean directory:

      stack unpack $package-$version # `-$version` is optional
      cd $package-$version
      rm -f stack.yaml && stack init --resolver nightly --ignore-subdirs
      stack build --resolver nightly --haddock --test --bench --no-run-benchmarks

ysangkok commented on build-constraints.yaml:
```diff
@@ -1620,27 +1620,6 @@ packages:
-        - optparse-simple
-        - hpack
-        - bindings-uname
-        - stack < 9.9.9 || > 9.9.9
```
@mpilgrem Does this need to get assigned to you?

mpilgrem replied:
I am happy for `stack` to be assigned to me - although I have questions over its suitability as a Stackage snapshot package; see https://github.com/commercialhaskell/stack/issues/6159#issuecomment-1596116151

In respect of `hpack`, I think @sol is the person to ask.

The others, I have had no involvement with the packages and `stack` already takes up most of my free time.

juhp commented:
@borsboom could you please move the packages to the Grandfathered or Unmaintained section instead.
We don't like to remove packages from Stackage "just because", thanks

borsboom commented:
@juhp Thanks for telling me about the "Grandfathered dependencies" section; I didn't know about that.  I've moved the packages there (except for `stack` which I moved to @mpilgrem's section).
